### PR TITLE
Docker env variable

### LINKED ISSUE #33: PUID and PGID support in container
**Is your feature request related to a problem? Please describe.**
 I run your container on unraid from this contributed template (https://forums.unraid.net/topic/122788-support-jpdvm2014-templates/), but from what I can see there is no way to define which PUID and PGID the application within the container should run as, and so it seems to run as root by default. This causes some permission issues with newly created paths that are used by other applications with non-root permissions.

**Describe the solution you'd like**
A similar way that linuxserver.io does where you define PUID= and PGID= with unique ID's that the application runs as within the container. This would resolve the issue with files being written as root

### PR DESCRIPTION
**Pull request by Berserkir-Wolf**
To resolve #33 (and #35 as a side-effect).

System will generate a user with sudo permissions inside the container, but with whatever UID and GID you define as environment variables (docs modification to follow later). Default UID/GID inside the container is 1000 (with this modification).

Note that this change does not break existing deployments providing the UID has permission to the directories in place (UID 1000 is the first non-root user created on a debian-based system by default).

### CODE REVIEW

**Comment by Casvt** [comment]
A few things:

1. Have you tested the PUID:PGID changes? Does it work?
2. In the docs there is an extra tab for if you want to specify PUID:PGID in the docker-compose file, but no instructions on how to do this for the Docker CLI command. 
3. All the changes you suggested in your other docs-PR are also made here, why?
4. The `<username>` found starting from [here](https://github.com/Casvt/Kapowarr/pull/55/files#diff-8b042e3f94ca5c59a7cd990b950aec0073ea84fe811e7b22be51158d7b180d56R156) don't display in the resulting docs. Install the `docs-requirements.txt` file and run `mkdocs serve` in a terminal to test your changes locally.
5. Remove the unzipping block

File: docs/installation.md [hunk 1]
```diff
@@ -46,6 +46,27 @@ The recommended way to install Kapowarr is using Docker. After installing Kapowa
     4. Information on how to change the port can be found on the [Setup After Installation page](./setup_after_installation.md#port).
     5. Using a named volume in docker requires you to create the volume before you can use it (refer to [Named Volumes](#named-volumes))
 
+=== "Docker CLI (custom user)"
+    The command to get the docker container running can be found below:
+    ```bash
+    docker run -d \
+        --name kapowarr \
+        -e PUID:1000 \
+        - GUID:1000 \
```
Comment (Casvt) [inline docs/installation.md]: Should be `-e`

**Comment by Berserkir-Wolf** [comment]
1. I have, it does.
2. Added in c24502b
3. I didn't switch back to the dev branch when creating this one - boss came out and distracted me at lunch.
4. Fixed in 09dca35
5. Gone in 3740aaf
6. If you set PUID/PGID this way, the application runs under a user with that ID inside the container. 
When it then saves/adjusts files, they get placed in the mounted volume as being attached to that same UID/GID, which has a different mapping in the host system.  
I have mediauser:mediagroup setup in my test setup, and when setting the IDs to match then the files show as being owned by the aforementioned group. It's a quick fix, in preparation for the new `-username` option that docker now has.
I say 'new' only because there are still relatively few references on how to switch a container to use it rather than the old PUID:PGID method. Not even linuxserver have switched over yet. As I slowly get my head around that one, I'll be modifying the dockerfile in my test environment to see if I can make it work there.

File: docs/installation.md (hunk 1 above)
Comment (Berserkir-Wolf) [reply docs/installation.md]: I was partway through writing the amendment to that one already.

**Comment by Berserkir-Wolf** [comment]
18146b0 fixed the 'building a library' conflict - only a merge because it was done in the web editor rather than VS Code.

**Comment by ggrensteiner** [comment]
If this has been merged it still doesn't work. PGID and PUID are ignored in docker-compose and the closest thing to docs about it give the wrong format for docker-compose

```
    environment:
     - PUID: 1000
     - PGID: 1000

```
is not valid syntax

```
    environment:
     - PUID=1000
     - PGID=1000
```
is ignored and the container runs as root and writes files as root.

**Comment by Casvt** [comment]
These changes have been removed for a long time because they indeed don't work. The current new PR #199 for this feature also doesn't actually work. Support for it is planned for V1.3.0.

**Pull request merged**